app/assets/javascripts/restful_typeahead.js

```diff
@@ -0,0 +1,220 @@
+		elemRef.attr('qryType',ttype);
+	}
+	if (!inid) {
+		//create a dynamic id if there is none
+		idcount += 1;
+		inid = 'srchInput_'+ttype+'_'+idcount;
+		elemRef.attr('id',inid);
+	}
+	//Check that there is a datalist element--if not, create one
+	var listid = elemRef.attr('data-results');
+	var listelem = $('#'+listid);
+	if (!listid) {
+		//create a datalist
+		listid = 'dlist_'+inid;
+		elemRef.attr('data-results',listid);
+		var listhtml = '<ul class="typeahead-results results-noshow" id="'+listid+'"></ul>';
+		listelem = $(listhtml).insertAfter(elemRef);
+	}
+	return listelem;
+}
+
+/**
+  Set up to perform search on key stroke entries for the input element.  If "Enter" or "Esc" is the key stroke, then hide the typeahead.
+**/
+function typeaheadSearchKeys(ielem, reslist) {
+        var elemRef = $(ielem);
+        var listelem = $(reslist);
+	elemRef.on("keyup", function(e) {
+		e.preventDefault();
+		var kcode = e.which || e.keyCode;
+		if (kcode == 13 || kcode == 27) {
+			//hit enter or escape, so hide the typeahead
+			listelem.toggleClass('results-noshow',true);
+		} else {
+			typeaheadSearch(this);
+		}
+	});
+}
+
+/**
+  Set up to perform tag search on key stroke entries for the input element.  If "Enter" or "Esc" is the key stroke, then hide the typeahead.
+**/
+function typeaheadTagKeys(ielem, reslist) {
+	var elemRef = $(ielem);
+	var listelem = $(reslist);
+	elemRef.on("keyup", function(e) {
+		e.preventDefault();
+		var kcode = e.which || e.keyCode;
+		if (kcode == 13 || kcode == 27) {
+			//hit enter or escape, so hide the typeahead
+			listelem.toggleClass('results-noshow',true);
+		} else {
+			typeaheadTags(this);
+		}
+	});
+
+}
+
+/**
+  On lost focus (blur) of the input element, hide the result list
+**/
+function hideResultsOnBlur(ielem, resList) {
+	var elemRef = $(ielem);
+	var listelem = $(resList);
+        elemRef.on("blur", function(e) {
+		listelem.toggleClass('results-noshow',true);
+	});
+}
+
+/**
+  Process the element's typed values and perform the query; display the results in the associated datalist
+**/
+function typeaheadSearch(srchElem) {
+	var qtype = $(srchElem).attr('qryType');
+	keycount += 1;
+	var qparams = new Object();
+	qparams.srchString = $(srchElem).val();
+	qparams.seq = keycount;
+	$.getJSON(typeaheadBase+qtype,qparams,function(qdata) {
+		if (qdata.srchParams) {
+			if (qdata.srchParams.seq >= keycount) {		
+				typeaheadTagList(srchElem, qdata);
+			}
+		}
+	});
+}
+
+/**
+  Handle the return values for a typeahead call which returns a list of document data
+**/
+function typeaheadDocList(ielem, docList) {
+	var elemRef = $(ielem);
+	var elemList = $('#'+elemRef.attr('data-results'));
+	elemList.html('');
+	if (docList.items) {
+		for (var i=0;i<docList.items.length;i++) {
+			var listopt = $('<li data-text="'+docList.items[i].docTitle+'"><span class="fa fa-'+docList.items[i].docType+'"></span>&nbsp;'+docList.items[i].docTitle+'</li>');
+			listopt.mousedown(function(e) {
```

Comment:
I tried the "click" method, but it caused the display and follow-up javascript calls to fail because the elements on the page were being manipulated by several methods at once--lots of "not found" or "undefined" errors.  Once I switched to mousedown, that all worked (and was the recommended way of handling this).

I will try to perform a test with a couple of my mobile devices (Android tablet and IPad/IPhone) to verify that this works properly.